halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/node/NodeReference.java

```diff
@@ -14,23 +14,19 @@
  * limitations under the License.
  */
 
-package com.netflix.spinnaker.halyard.config.model.v1;
+package com.netflix.spinnaker.halyard.config.model.v1.node;
 
 import lombok.Data;
-import lombok.EqualsAndHashCode;
-import lombok.NoArgsConstructor;
 
 /**
- * A reference to a field in halconfig.
- * @param <T> The type of the update
+ * A way to identify a spot in your halconfig.
  */
 @Data
-@NoArgsConstructor
-@EqualsAndHashCode(callSuper = false)
-public class FieldReference<T> extends Reference<T> {
-
-  /**
-   * The name of the field being updated.
-   */
-  private String fieldName;
+public class NodeReference implements Cloneable {
```

Comment:
Chatted offline, the idea is that since we dictate the structure this is OK for now, but worth revisiting. (The nice thing about the current abstraction is that new providers can come in and make changes without touching this model, so we are free to refactor/extend it).